pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1049,6 +1069,39 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 		}
 	}
 
+	// TODO where to store harcoded strings
+	if env.Mesh.GetEnableAutoMtls().Value {
```

Comment:
@diemtvu I think we should only set `transport_socket_matches` when server side is in permissive mode, not unconditionally. Plus today we have a plaintext mode even...

We should add a check of the server side policy from push context.